include/boost/geometry/formulas/distance_point_segment.hpp

```diff
@@ -0,0 +1,345 @@
+// Boost.Geometry
+
+// Copyright (c) 2016-2017 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_CROSS_TRACK_GEO_HPP
+#define BOOST_GEOMETRY_FORMULAS_CROSS_TRACK_GEO_HPP
+
+#include <boost/geometry/formulas/result_direct.hpp>
+
+#ifndef BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS
+#define BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS 100
+#endif
+
+/*!
+\brief Algorithm to compute the distance between a segment and a point using
+       direct and inverse geodesic problems as subroutines. The algorithm
+       approximates the distance by an iterative Newton method.
+\see C.F.F.Karney - Geodesics on an ellipsoid of revolution,
+      https://arxiv.org/abs/1102.1215
+*/
+
+namespace boost { namespace geometry { namespace formula
+{
+
+template
+<
+        typename CT,
+        typename Units,
+        template <typename, bool, bool, bool, bool ,bool> class Inverse,
+        template <typename, bool, bool, bool, bool> class Direct,
+        bool EnableClosestPoint = false
+>
+class distance_point_segment{
+
+public:
+
+    typedef Inverse<CT, true, false, false, true, true> inverse_distance_type;
```

Comment:
Here the name doesn't fully reflect the parameters since besides distance quantities are calculated but I guess that's fine.